### PR TITLE
[ci] fix base image repo mapping

### PR DESCRIPTION
**Pull request by aslonnie**
also added unit tests on base image generation.

### CODE REVIEW

File: release/ray_release/test.py [hunk 1]
```diff
@@ -276,14 +279,16 @@ def get_ray_image(self) -> str:
...
+            repo_name = config["byod_ray_cr_repo"]
+        elif repo == DATAPLANE_ECR_ML_REPO:
+            repo_name = config["byod_ray_ml_cr_repo"]
+        else:
+            raise ValueError(f"Unknown repo {repo}")
+
         return (
-            f"{config['byod_ray_ecr']}/"
-            f"{ray_project}:{self.get_byod_base_image_tag()}"
+            f"{config['byod_ray_ecr']}/" f"{repo_name}:{self.get_byod_base_image_tag()}"
```
Comment (xwjiang2010) [inline release/ray_release/test.py:291]: this can just be one line?

Comment (aslonnie) [reply release/ray_release/test.py]: done.

**Review by xwjiang2010 — APPROVED** [review]
rubber stamping

**Review by scv119 — APPROVED** [review]
🐶 🧑‍🔬

**Comment by can-anyscale** [comment]
Thanks for fixing!

**Pull request merged**